doc/rst/language/spec/statements.rst

```diff
@@ -533,6 +553,21 @@ termination expression.
       4
       5
 
+The control-flow declaration *ctrl-decl*, when used in a while-do loop,
+works similarly to how it does in a conditional statement.
+It declares a variable whose scope is the loop body.
+Its *expression* must be of a class type and is considered to evaluate
```

Comment:
It's not the "expression" that evaluates to `true`, right?  Because then the identifier would store `true` rather than a class instance.  I think you probably want to say "the control-flow declaration is considered to evaluate to ``true``" instead